sds_data_manager/constructs/instrument_lambdas.py

```diff
@@ -0,0 +1,67 @@
+"""Module containing Constructs for instsrument Lambda functions"""
+from pathlib import Path
+
+from aws_cdk import Duration
+from aws_cdk import aws_lambda as lambda_
+from aws_cdk import aws_lambda_python_alpha as lambda_alpha_
+from aws_cdk import aws_s3 as s3
+from constructs import Construct
+
+
+class InstrumentLambda(Construct):
+    """Generic Construct with customizable runtime code
+    """
+
+    def __init__(self,
+                 scope: Construct,
+                 construct_id: str,
+                 sds_id:str,
+                 processing_step_name: str,
+                 archive_bucket: s3.Bucket,
+                 code_path: str or Path,
+                 instrument_target: str,
+                 instrument_sources: str):
+        """InstrumentLambda Constructor
+
+        Parameters
+        ----------
+        scope : Construct
+        construct_id : str
+        sds_id : str
+            Name suffix for stack
+        processing_step_name : str
+            Processing step name
+        archive_bucket: s3.Bucket
```

Comment:
Can you give a background of why we are using `archive_bucket` in these constructs? Just curious. I feel like data that we are using in processing pipeline should be coming from the data bucket itself. But in the architecture diagram, we are sending data to database(Eg. opensearch) and to archival bucket. So thought you are passing that argument for that. Is that right?

Reply:
Yep. We discussed this. It should be the same bucket. Sorry for the confusion.